src/ape/api/transactions.py

```diff
@@ -147,22 +148,25 @@ class ReceiptAPI(BaseInterfaceModel):
 
     contract_address: Optional[str] = None
     block_number: int
-    data: bytes = b""
-    gas_limit: int
-    gas_price: int
     gas_used: int
     logs: List[dict] = []
-    nonce: Optional[int] = None
-    receiver: str
-    required_confirmations: int = 0
-    sender: str
     status: int
     txn_hash: str
-    value: int = 0
+    transaction: TransactionAPI
 
     def __repr__(self) -> str:
         return f"<{self.__class__.__name__} {self.txn_hash}>"
 
+    def __getattr__(self, item: str) -> Any:
+        return getattr(self.transaction, item)
+
+    @validator("transaction", pre=True)
+    def confirm_transaction(cls, value):
+        if isinstance(value, dict):
+            value = TransactionAPI.parse_obj(value)
```

Comment:
This class will have unimplemented API methods, no?

Reply:
I originally had that isinstance in there for when we built dictionaries for tests. This might be better off being removed. I'm honestly not so sure we need it anymore after all the changes that were made to the tests. If you give TransactionAPI the data it is looking for, it is able to parse and build a transaction though. I rigorously tested it. It is not currently being tested now. I will make a ticket to add a test for this section of the validator to make sure this works correctly.